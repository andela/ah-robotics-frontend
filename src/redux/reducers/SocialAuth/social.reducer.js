import {
  SOCIAL_LOGIN_ERROR,
  SOCIAL_LOGIN_START,
  SOCIAL_LOGIN_SUCCESS,
  } from '../../actions/socialAuth/types';

export const initialState = {
  isLoading: false,
  error: false,
};

const SocialAuthReducer = (state = initialState, action) => {
  switch (action.type) {
    case SOCIAL_LOGIN_START:
      return {
        ...state,
        ...action.payload,
        isLoading: true,
        message: 'fetching user details',
        error: false,
      };
    case SOCIAL_LOGIN_SUCCESS:
      return {
        ...state,
        ...action.payload,
        message: 'success',
        isLoading: false,
        error: false,
        errors: {},
      };
    case SOCIAL_LOGIN_ERROR:
      return {
        ...action.payload,
        message: 'failed',
        isLoading: false,
        error: true,
      };

    default:
      return state;
  }
};

export default SocialAuthReducer;
