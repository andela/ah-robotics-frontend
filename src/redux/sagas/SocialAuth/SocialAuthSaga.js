import { put, call, takeEvery } from 'redux-saga/effects';
import * as types from '../../actions/socialAuth/types';
import setUserToken, { api, setCurrentUser } from '../../../utils/auth';

const apiUrl = '/login/social/';

export const apiClient = {
  fetchSocialLoginOptions: (payload) => {
    const { authData } = payload;
    return api.post(apiUrl, authData);
  },
};

export function* SocialAuthSaga({ payload }) {
  try {
    const { userDetails } = payload;
    const response = yield call(apiClient.fetchSocialLoginOptions, payload);
    yield put({
      type: types.SOCIAL_LOGIN_SUCCESS,
      payload: { user: response.data.user },
    });
    const user = {
      email: userDetails.email,
      username: (userDetails.email).match(/^([^@]*)@/)[1],
    };
    setUserToken(response.data.token);
    setCurrentUser(user);
  } catch (error) {
    yield put({
      type: types.SOCIAL_LOGIN_ERROR,
      payload: { errors: error.response ? error.response.data.errors : {} },
    });
  }
}

function* socialAuthWatcher() {
  yield takeEvery(types.SOCIAL_LOGIN_START, SocialAuthSaga);
}

export default socialAuthWatcher;
