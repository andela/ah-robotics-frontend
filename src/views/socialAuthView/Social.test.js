import React from 'react';

import SocialAuth from '.';
import ButtonComponent from '../../components/common/SocialButton';

function setup() {
  const social = {};
  const handleSubmit = () => ({});
  const onChange = () => ({});

  const wrapper = shallow(
    <SocialAuth handleSubmit={handleSubmit} social={social} onChange={onChange} />,
);

  return {
    wrapper,
  };
}

describe('Social View', () => {
  it('should render self and subcomponents', () => {
    const { wrapper } = setup();
    expect(wrapper.find('button')).toBeDefined();
  });
  it('app should match the snapshot', () => {
    const wrapper = shallow(<SocialAuth />);
    expect(wrapper).toMatchSnapshot();
  });
});

it('app should render social login view page', () => {
  const wrapper = shallow(<SocialAuth />);
  expect(wrapper.length).toBe(1);
});


describe('test create article view', () => {
it('it render create article view', () => {
const wrapper = shallow(<SocialAuth />);
expect(wrapper.length).toBe(1);
});

it('it render social buttons', () => {
  const wrapper = shallow(<ButtonComponent />);
  expect(wrapper.length).toBe(1);
  });

it('it should match snapshot', () => {
  const wrapper = shallow(<SocialAuth />);
  expect(wrapper).toMatchSnapshot();
});
});
