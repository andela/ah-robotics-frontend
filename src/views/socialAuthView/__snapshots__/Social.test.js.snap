// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social View app should match the snapshot 1`] = `
<Route>
  <Component />
</Route>
`;

exports[`test create article view it should match snapshot 1`] = `
<Route>
  <Component />
</Route>
`;
